www-src/app/replicator/change/change_dispatcher.coffee

```diff
@@ -33,14 +34,10 @@ module.exports = class ChangeDispatcher
      * @param {Object} doc - it's a pouchdb file document.
     ###
     dispatch: (doc, callback = ->) ->
-        log.info "change #{doc.docType}"
-        if doc.docType is "file"
-            state = @_getState doc
-            @changeHandlers.file[state] doc
+        log.info "dispatch #{doc.docType}"
 
-        handler = @changeHandlers[doc.docType]
-        if handler?
-            handler.dispatch doc, callback
+        if @isDispatched doc and @changeHandlers[doc.docType]
```

Comment:
Be careful with coffee priority in boolean computation : it's really hard to know what coffee will do. the joseph's way is to ALWAYS use parenthesis on arguments in boolean computation, and in this case it was needed: 
```
if @isDispatched(doc) and @changeHandlers[doc.docType]
```